lib/vmfloaty.rb

```diff
@@ -63,9 +64,13 @@ def run # rubocop:disable Metrics/AbcSize
           exit 1
         end
 
-        response = service.retrieve(verbose, os_types, use_token)
+        response = service.retrieve(verbose, os_types, use_token, options.ondemand)
+        request_id = response['request_id'] if options.ondemand
+        response = service.wait_for_request(verbose, request_id) if options.ondemand
```

Comment:
These should probably be wrapped in a single if-statement for readability.

Reply:
Done.